net/DevExtreme.AspNet.Data/DataSourceLoader.cs

```diff
@@ -76,10 +73,20 @@ public class DataSourceLoader {
             return result;
         }
 
+        static IEnumerable EnsureBuffered<T>(IEnumerable data) {
```

Comment:
I would suggest renaming this method to `Evaluate`, or `Buffer` or smth like. As for me `Ensure` supposes some internal changes and no value returned.